### PR TITLE
Take a generic host instead of using a port parameter if the different port is needed

### PR DESCRIPTION
**Pull request by daenamkim**
In some case, a developer might use a different port for localhost(or deployment) so adding a `deploymentPort` parameter for Provider and deploying contract

### CODE REVIEW

**Review by shoenseiwaso — COMMENTED** [review]
Good call - can we actually just make this more generic, and switch to accepting a deployment URL instead of a deployment ID and (optionally) a deployment port? Similar to the spreadsheet plugin. I think it will give us the greatest flexibility going forward.

**Comment by daenamkim** [comment]
@shoenseiwaso I hope you take a look at this PR when you are available.

File: src/config.ts [hunk 1]
```diff
@@ -80,6 +79,11 @@ export default function getConfig(): Config {
  * @param deploymentID The deployment ID in the config.
  */
 export function getHost(deploymentID: string): string {
+  const re = /^http.*:\/\/[a-zA-Z0-9\-._]+:[0-9]+/;
```
Comment (shoenseiwaso) [inline src/config.ts:82]: Do you have a source for the this regex? Want to make sure we're catching all the corner cases.

Comment (daenamkim) [reply src/config.ts:82]: I just tested this from https://regex101.com/ with:

- http://localhost:8080
- https://localhost:8081

**Comment by daenamkim** [comment]
@shoenseiwaso I have tested as follows:

```js
// { insecureOk: false }
console.log("1.1.1.1", isURL("1.1.1.1", validatorOptions));
console.log("1.1.1.1:8080", isURL("1.1.1.1:8080", validatorOptions));
console.log("http://1.1.1.1", isURL("http://1.1.1.1", validatorOptions));
console.log("http://1.1.1.1:8080", isURL("http://1.1.1.1:8080", validatorOptions));
console.log("localhost", isURL("localhost", validatorOptions));
console.log("localhost:8080", isURL("localhost:8080", validatorOptions));
console.log("http://localhost", isURL("http://localhost", validatorOptions));
console.log("http://localhost:8080", isURL("http://localhost:8080", validatorOptions));
console.log("google.com", isURL("google.com", validatorOptions));
console.log("index.hu", isURL("index.hu", validatorOptions));
console.log("www.index.hu", isURL("www.index.hu", validatorOptions));
console.log("http://www.index.hu", isURL("http://www.index.hu", validatorOptions));
console.log("https://www.index.hu", isURL("https://www.index.hu", validatorOptions)); // success

// { insecureOk: true }
console.log("1.1.1.1", isURL("1.1.1.1", validatorOptions));
console.log("1.1.1.1:8080", isURL("1.1.1.1:8080", validatorOptions));
console.log("http://1.1.1.1", isURL("http://1.1.1.1", validatorOptions)); // success
console.log("http://1.1.1.1:8080", isURL("http://1.1.1.1:8080", validatorOptions)); // success
console.log("localhost", isURL("localhost", validatorOptions));
console.log("localhost:8080", isURL("localhost", validatorOptions));
console.log("http://localhost", isURL("http://localhost", validatorOptions)); // success
console.log("http://localhost:8080", isURL("http://localhost:8080", validatorOptions)); // success
console.log("google.com", isURL("google.com", validatorOptions));
console.log("index.hu", isURL("index.hu", validatorOptions));
console.log("www.index.hu", isURL("www.index.hu", validatorOptions));
console.log("http://www.index.hu", isURL("http://www.index.hu", validatorOptions)); // success
console.log("https://www.index.hu", isURL("https://www.index.hu", validatorOptions)); // Success
```

**Review by shoenseiwaso — APPROVED** [review]
LGTM!

**Pull request merged**